lib/pgBackRest/Config/Config.pm

```diff
@@ -18,2089 +18,103 @@ use pgBackRest::Common::Exception;
-        &OPTION_RULE_ALLOW_LIST =>
-        {
-            lc(OFF)    => true,
-            lc(ERROR)  => true,
-            lc(WARN)   => true,
-            lc(INFO)   => true,
-            lc(DETAIL) => true,
-            lc(DEBUG)  => true,
-            lc(TRACE)  => true
-        },
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD_CHECK => true,
-            &CMD_EXPIRE => true,
-            &CMD_INFO => true,
-            &CMD_RESTORE => true,
-            &CMD_STANZA_CREATE => true,
-            &CMD_STANZA_UPGRADE => true,
-            &CMD_START => true,
-            &CMD_STOP => true
-        }
-    },
-
-    &OPTION_LOG_LEVEL_FILE =>
-    {
-        &OPTION_RULE_SECTION => CONFIG_SECTION_GLOBAL,
-        &OPTION_RULE_TYPE => OPTION_TYPE_STRING,
-        &OPTION_RULE_DEFAULT => OPTION_DEFAULT_LOG_LEVEL_FILE,
-        &OPTION_RULE_ALLOW_LIST =>
-        {
-            lc(OFF)    => true,
-            lc(ERROR)  => true,
-            lc(WARN)   => true,
-            lc(INFO)   => true,
-            lc(DEBUG)  => true,
-            lc(DETAIL) => true,
-            lc(TRACE)  => true
-        },
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD_CHECK => true,
-            &CMD_EXPIRE => true,
-            &CMD_INFO => true,
-            &CMD_RESTORE => true,
-            &CMD_STANZA_CREATE => true,
-            &CMD_STANZA_UPGRADE => true,
-            &CMD_START => true,
-            &CMD_STOP => true
-        }
-    },
-
-    &OPTION_LOG_LEVEL_STDERR =>
-    {
-        &OPTION_RULE_SECTION => CONFIG_SECTION_GLOBAL,
-        &OPTION_RULE_TYPE => OPTION_TYPE_STRING,
-        &OPTION_RULE_DEFAULT => OPTION_DEFAULT_LOG_LEVEL_STDERR,
-        &OPTION_RULE_ALLOW_LIST =>
-        {
-            lc(OFF)    => true,
-            lc(ERROR)  => true,
-            lc(WARN)   => true,
-            lc(INFO)   => true,
-            lc(DETAIL) => true,
-            lc(DEBUG)  => true,
-            lc(TRACE)  => true
-        },
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD_CHECK => true,
-            &CMD_EXPIRE => true,
-            &CMD_INFO => true,
-            &CMD_RESTORE => true,
-            &CMD_START => true,
-            &CMD_STOP => true
-        }
-    },
-
-    &OPTION_LOG_TIMESTAMP =>
-    {
-        &OPTION_RULE_SECTION => CONFIG_SECTION_GLOBAL,
-        &OPTION_RULE_TYPE => OPTION_TYPE_BOOLEAN,
-        &OPTION_RULE_DEFAULT => OPTION_DEFAULT_LOG_TIMESTAMP,
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD
```

Comment:
It appears that some constants in Config.pm are "duplicated" in Data.pm. (e.g. REPO_TYPE_x == CFGBLDOPTVAL_REPO_TYPE_x, INFO_OUTPUT_x == CFGBLDOPTVAL_INFO_OUTPUT_x, BACKUP_TYPE_x == CFGBLDOPTVAL_BACKUP_TYPE_x)
Will this always be the case and if so, how do we determine what needs to be duplicated and what does not?